### PR TITLE
[stable28]  fix: Adjust reference picker code for the vue based FilePicker

### PR DESCRIPTION
**Pull request by susnux**
Manual backport of #https://github.com/nextcloud/server/pull/42104

### CODE REVIEW

File: apps/files/src/views/FileReferencePickerElement.vue [hunk 1]
```diff
@@ -40,74 +49,55 @@ export default {
...
-				false, // modal
-				FilePickerType.Choose, // type
-				'',
-				{
-					target: this.$refs.picker,
-				},
-			)
+
+		onClose(nodes?: NcNode[]) {
+			if (!nodes) {
```
Comment (Pytal) [inline apps/files/src/views/FileReferencePickerElement.vue:88]:
```suggestion
			if (nodes.length === 0) {
```
Arrays are always truthy

**Review by Pytal — APPROVED** [review]

File: apps/files/src/views/FileReferencePickerElement.vue (hunk 1 above)
Comment (blizzz) [reply apps/files/src/views/FileReferencePickerElement.vue:88]: Mh, then it should be fixed (also on master), for then it must be true and the other condition could never trigger? Or do I misinterpret something, as you'd wave it through with the checkmark?

Comment (Pytal) [reply apps/files/src/views/FileReferencePickerElement.vue:88]: A prioritized RC3 checkmark rather in anticipation of a quick fix 🙈

Comment (blizzz) [reply apps/files/src/views/FileReferencePickerElement.vue:88]: But it's the last RC before the final, there will be no possibility for a fix anymore, unless something is terribly broken. So either we can live with that, or fix it before merging.

Comment (Pytal) [reply apps/files/src/views/FileReferencePickerElement.vue]: Quick fixed :)

Comment (blizzz) [reply apps/files/src/views/FileReferencePickerElement.vue]: Thank you! =)

Comment (Pytal) [reply apps/files/src/views/FileReferencePickerElement.vue]: Actually the previous quick fix is also not 100% correct as the code reveals some unintuitive behaviour https://github.com/search?q=repo%3Anextcloud-libraries%2Fnextcloud-dialogs%20%22emit(%27close%27%22&type=code which should be handled explicitly

Pushed the real fix

Comment (Pytal) [reply apps/files/src/views/FileReferencePickerElement.vue]: In general I'd say we should use `null` rather than `undefined` for explicitness @susnux

Comment (blizzz) [reply apps/files/src/views/FileReferencePickerElement.vue]: and the fixes should go to the master PR as well

**Review by blizzz — APPROVED** [review]

**Pull request merged**